rfd/0134-cloud-first-deployment.md

```diff
@@ -0,0 +1,203 @@
+of the current release process.
+
+Current release versioning scheme follows semver, with quarterly major releases
+and monthly minor releases.
+
+Major releases are getting rolled onto Cloud quarterly, with a month's delay.
+Major features are reserved for major releases, which leads to Cloud customers
+getting major features with a month's delay.
+
+The release process is triggered on a GitHub tag creation and builds the full
+suite of artifacts that takes 2-3 hours so it's unrealistic to do multiple
+deploys a day.
+
+Cloud upgrade happens once a week to a release published the previous week, and
+engineers oftentimes don't have a clear idea of when their change is being
+deployed.
+
+## Cloud release process
+
+In the Cloud-first world, every release that's deployed on Cloud does not necessarily
+have a corresponding self-hosted release tag. For this reason, we can't reuse
+the current versioning scheme and apply it to both Cloud and self-hosted.
+
+Instead, we want to get to a place where Cloud and self-hosted releases are
+decoupled, with Cloud releases being regularly cut from the main development
+branch and self-hosted releases cut from their respective release branches.
+
+Although the concept of semantic versioning loses most of its meaning with the
+continuous release train, we do need to keep it for Cloud releases to be able
+to use automatic upgrades system we built.
+
+With this in mind, the Cloud version will consist of:
+
+- Major: "development" version number on master branch (latest stable self-hosted release + 1).
+- Minor/patch: always 0.
+- Prerelease: `-cloud.X` with a perpetually increasing "build" number which will
+  provide enough semver comparison guarantees for auto-upgrades to work.
+
+Once it is time for the next self-hosted major release (every 3-4 months), the
+new release branch will cut and the Cloud release version major component will
+increment.
+
+```
+                                14.0.0     14.0.1     14.0.2
+                     branch/v14 |----------|----------|---------....---->
+                                |
+                                |
+            14.0.0-cloud.1      14.0.0-cloud.10    15.0.0-cloud.0     15.0.0-cloud.1     16.0.0-cloud.0
+master -----|-------....--------|------------------|-------....-------|------------------|--------------....---->
+                                                                      |
+                                                                      |
+                                                           branch/v15 |---------|---------|---------...---->
+                                                                      15.0.0    15.0.1    15.0.2
+```
+
+Once a self-hosted release branch is cut, subsequent self-hosted minor/patch
+releases are published without any changes to the current process, but on a
+less frequent cadence than Cloud releases off of master.
+
```

Comment:
Will changes to the self-hosted release branches generally follow the same develop-on-master/backport-to-branch workflow we're using now?

Reply:
Yep, added a sentence to clarify this.